src/site/content/en/reliable/two-way-communication-guide/index.md

```diff
@@ -0,0 +1,400 @@
+context:
+
+```javascript
+//send message
+broadcast.postMessage({ type: 'MSG_ID', });
+```
+
+Any browser context can listen to messages via the `onmessage` method of the `BroadcastChannel`
+object:
+
+```javascript
+//listen to messages
+broadcast.onmessage = (event) => {
+  if (event.data && event.data.type === 'MSG_ID') {
+    //process message...
+  }
+};
+```
+
+As seen, there's no explicit reference to a particular context, so there's no need of obtaining a
+reference first to the service worker or any particular client.
+
+<figure class="w-figure">
+  <img src="broadcast-channel.png"
+       width="700"
+       alt="Diagram showing two-way communication between page and service worker, using a Broadcast Channel object.">
+</figure>
+
+The disadvantage is that, at the moment of this writing, the API has support from Chrome, Firefox
+and Edge, but other browsers, like Safari, [don't support it
+yet](https://caniuse.com/?search=broadcastchannel).
+
+### Client API
+
+The [Client API](https://developer.mozilla.org/en-US/docs/Web/API/Client) allows you to obtain a
+reference to all the [WindowClient
+objects](https://developer.mozilla.org/en-US/docs/Web/API/WindowClient) representing the active tabs
+that the service worker is controlling.
+
+Since the page is controlled by a single service worker, it listens to and sends messages to the
+active service worker directly via the `serviceWorker` interface:
+
+```javascript
+//send message
+navigator.serviceWorker.controller.postMessage({
+  type: 'MSG_ID',
+});
+
+//listen to messages
+navigator.serviceWorker.onmessage = (event) => {
+  if (event.data && event.data.type === 'MSG_ID') {
+    //process response
+  }
+};
+```
+
+Similarly, the service worker listens to messages by implementing an `onmessage` listener:
+
+```javascript
+//listen to messages
+self.addEventListener('message', (event) => {
+  if (event.data && event.data.type === 'MSG_ID') {
+    //Process message
+  }
+});
+```
+
+To communicate back with any of its clients, the service worker obtains an array of
+[WindowClient](https://developer.mozilla.org/en-US/docs/Web/API/WindowClient) objects by executing
+methods such as
+[Clients.matchAll()](https://developer.mozilla.org/en-US/docs/Web/API/Clients/matchAll) and
+[Clients.get()](https://developer.mozilla.org/en-US/docs/Web/API/Clients/get). Then it can
+`postMessage()` any of them:
+
+```javascript
+//Obtain an array of Window client objects
+self.clients.matchAll(options).then(function (clients) {
+  if (clients && clients.length) {
+    //Respond to last focused tab
+    clients[0].postMessage({type: 'MSG_ID'});
+  }
+});
+```
+
+<figure class="w-figure">
+  <img src="client-api.png"
+       width="500"
+       alt="Diagram showing a service worker communicating with an array of clients.">
+</figure>
+
+`Client API` is a good option to communicate easily with all the active tabs from a service worker
```

Comment:
Formatting of Client API isn't consistent with earlier

Reply:
I see in all cases it appears as `Client API`. I can't find the mismatch here.